### ISSUE TITLE
Subword matching fails on upper case sequences

### ISSUE THREAD

**Issue by brxck** [pokey/cursorless-vscode#79]
This is a bug knowingly introduced by #67 

Subwords in camel/parcal-cased tokens with a sequence of upper case letters are not distinguished correctly. Anything after a sequence of upper case letters will register as one word.

Visualization of current behavior:

```
APIMatcherFactory -> APIMatcherFactory
```

vs desired behavior:

```
APIMatcherFactory -> API Matcher Factory
```

**Comment by AndreasArvidsson** [comment]
Have a look at the regexp I made for the reformatter:
https://github.com/knausj85/knausj_talon/blob/12229e932d9d3de85fa2f9d9a7c4f31ed6b6445b/code/formatters.py#L270

Specificly:
`
(?<=[A-Z])(?=[A-Z][a-z])
`

**Comment by pokey** [comment]
@AndreasArvidsson sorry my lookahead regex syntax is a bit rusty 😅.  Would that be the whole regex?  I don't see any repeats in there

**Comment by AndreasArvidsson** [comment]
That is not the whole regex. Check the link.

That part just matches where to break when there are sequiential upper case characters.

**Comment by pokey** [comment]
So do we wanna just use that whole regex then?

**Comment by AndreasArvidsson** [comment]
I have to look up how it's used first. This regular expression matches where to break not the tokens itself. So I'm going to have to do some modifying probably.

**Comment by AndreasArvidsson** [comment]
PR #88

**Comment by pokey** [comment]
closed by #88
